explorer/client/src/components/ownedobjects/OwnedObjects.tsx

```diff
@@ -33,174 +32,161 @@ const DATATYPE_DEFAULT: resultType = [
     },
 ];
 
-function OwnedObjectStatic({ objects }: { objects: string[] }) {
-    const results = objects.map((objectId) => {
-        const entry = findDataFromID(objectId, undefined);
-        return {
-            id: entry?.id,
-            Type: entry?.objType,
-            Version: entry?.version,
-            display: entry?.data?.contents?.display,
-        };
-    });
-
-    return <OwnedObjectView results={results} />;
+const IS_COIN_TYPE = (typeDesc: string): boolean => /::Coin::/.test(typeDesc);
+
+function OwnedObject({ id }: { id: string }) {
+    if (process.env.REACT_APP_DATA === 'static') {
+        return <OwnedObjectStatic id={id} />;
+    } else {
+        return <OwnedObjectAPI id={id} />;
+    }
 }
 
-function OwnedObjectAPI({ objects }: { objects: string[] }) {
+function OwnedObjectStatic({ id }: { id: string }) {
+    const objects = findOwnedObjectsfromID(id);
+
+    if (objects) {
+        const results = objects?.map(({ objectId }) => {
+            const entry = findDataFromID(objectId, undefined);
+            return {
+                id: entry?.id,
+                Type: entry?.objType,
+                Version: entry?.version,
+                display: entry?.data?.contents?.display,
+                balance: entry?.data?.contents?.balance,
+            };
+        });
+
+        return <GroupView results={results} />;
+    } else {
+        return <div />;
+    }
+}
+
+function OwnedObjectAPI({ id }: { id: string }) {
     const [results, setResults] = useState(DATATYPE_DEFAULT);
     const [isLoaded, setIsLoaded] = useState(false);
 
+    // TODO - The below will fail for a large number of owned objects
```

Comment:
This works pretty well in practice. This address(`813ec1ab1e1797c79d1e1fcbebebc27a1fd21f07`) owns 110+ coins and the page loads well

Reply:
if it can handle 110 objects, that's good enough for now i think.